### PR TITLE
BATIAI-2164-cross-alb: Need to prevent applications from cross-load balancer access

### PR DESCRIPTION
**Pull request by afaras0572**
## Description:
Need to prevent applications from cross-load balancer access

### Submitter Checklist
-  [ ] Is there an impact on Auditing and Logging procedures or capabilities?
-  [ ] Is there an impact on Authentication procedures or capabilities?
-  [ ] Is there an impact on Authorization procedures or capabilities?
-  [ X] Is there an impact on Communication Security procedures or capabilities?
-  [ ] Is there an impact on Cryptography procedures or capabilities?
-  [ ] Is there an impact on Sensitive Data procedures or capabilities?
-  [ ] Is there an impact on any other security-related procedures or capabilities?
-  [ ] No security impacts identified.

Security Risks Identified - For any applicable items on the "Submitter Checklist," describe the impact of the change and any implemented mitigations.

### CODE REVIEW

**Comment by robo-gotham** [comment]
Snyk Scanning for Commit: 14752a47fb7b29eb7b9cbc8343224cd623189900

Snyk Infrastructure as Code

- Snyk testing Infrastructure as Code configuration issues.
✔ Test completed.

Issues
  No vulnerable paths were found!

-------------------------------------------------------

</details>

Test Summary

  Organization: batcave-ispg
  Project name: CMS-Enterprise/batcave-tf-cluster

✔ Files without issues: 10
✗ Files with issues: 0
  Ignored issues: 0
  Total issues: 0 [ 0 critical, 0 high, 0 medium, 0 low ]

-------------------------------------------------------

Report Complete

  Your test results are available at: https://snyk.io/org/batcave-ispg/projects
  under the name: CMS-Enterprise/batcave-tf-cluster

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,3 +3,6 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+BATIAI-2164-cross-alb:
```
Comment (j-mowery) [inline CHANGELOG.md:7]: Do not directly modify the Changelog file

**Review by j-mowery — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (afaras0572) [reply CHANGELOG.md]: comment in change log file removed

**Review by j-mowery — APPROVED** [review]

**Review by bushong1 — CHANGES_REQUESTED** [review]
Gotta run the code formatter.  Looks good otherwise.

File: alb-private-to-k8s.tf [hunk 2]
```diff
@@ -27,22 +27,54 @@ resource "aws_lb" "batcave_alb" {
...
 resource "aws_lb_listener" "batcave_alb_https" {
   load_balancer_arn = aws_lb.batcave_alb.arn
   port              = "443"
   protocol          = "HTTPS"
   ssl_policy        = var.alb_ssl_security_policy
-  default_action {
-    type             = "forward"
-    target_group_arn = aws_lb_target_group.batcave_alb_https.arn
+  dynamic "default_action" {
+    for_each=length(var.alb_restricted_hosts) == 0 ?["forward all request"] : []
```
[the remark this answers is not in the document]
Comment (afaras0572) [reply alb-private-to-k8s.tf]: formatting has been done

**Review by bushong1 — APPROVED** [review]

**Pull request merged**